plugins/Aka/plugin.py

```diff
@@ -376,6 +396,104 @@ def findBiggestAt(alias):
 else:
     raise plugins.NoSuitableDatabase(['sqlite3', 'sqlalchemy'])
 
+
+class AkaHTTPCallback(httpserver.SupyHTTPServerCallback):
+    name = 'Aka web interface'
+    base_template = '''\
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Aka</title>
+    <link rel="stylesheet" href="/default.css">
+  </head>
+
+  <body>
+    <h1>Aka</h1>
+
+    {}
+  </body>
+</html>'''
+    index_template = base_template.format('''\
+    <p>To view the global Akas either click <a href="list/global">here</a> or
+    enter 'global' in the form below.</p>
+
+    <form action="" method="post">
+        <label for="channel">Channel name:</label>
+        <input type="text" placeholder="#channel" name="channel" id="channel">
+        <input type="submit" name="submit" value="view">
+    </form>''')
+    list_template = base_template.format('''\
+    <table>
+      <thead>
+        <th>Name</th>
+        <th>Alias</th>
+        <th>Locked</th>
+      </thead>
+      {}
+    </table>''')
+
+    def doGet(self, handler, path, *args, **kwargs):
+        if path == '/':
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html; charset=utf-8')
+            self.end_headers()
+            self.write(self.index_template)
+
+        elif path.startswith('/list/'):
+            parts = path.split('/')
+            channel = parts[2] if len(parts) == 3 else 'global'
+            channel = utils.web.urlunquote(channel)
+
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html; charset=utf-8')
+            self.end_headers()
+
+            akas = {}
+            for aka in self._plugin._db.get_all('global'):
+                akas[aka[1]] = aka
+
+            if channel != 'global':
+                for aka in self._plugin._db.get_all(channel):
+                    akas[aka[1]] = aka
+
+            aka_rows = []
+            for name in sorted(akas.keys()):
+                (id, name, alias, locked, locked_by, locked_at) = akas[name]
+                locked_column = 'False'
+                if locked:
+                    locked_column = 'By {} at {}'.format(
+                        locked_by,
+                        locked_at.split('.')[0],
```

Comment:
`utils.str.timestamp(float(locked_at))`

Reply:
They're stored as strings like `2019-10-05 22:40:03.531491` in SQLite3 though.

Reply:
oh, ok

Reply:
It is probably worth looking into passing `sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES` as the `detect_types` argument to `sqlite3.connect` option to get back an actual `datetime` object, but that should really be done as a separate thing.